llama_index/vector_stores/cassandra.py

```diff
@@ -0,0 +1,248 @@
+"""Cassandra / Astra DB Vector store index.
+
+An index based on a DB table with vector search capabilities,
+powered by the cassIO library
+
+"""
+
+import logging
+from typing import Any, List, Optional, cast
+
+
+from llama_index.vector_stores.utils import (
+    DEFAULT_TEXT_KEY,
+    metadata_dict_to_node,
+    node_to_metadata_dict,
+)
+from llama_index.indices.query.embedding_utils import (
+    get_top_k_mmr_embeddings,
+)
+
+from llama_index.vector_stores.types import (
+    # MetadataFilters,
+    NodeWithEmbedding,
+    VectorStore,
+    VectorStoreQuery,
+    VectorStoreQueryMode,
+    VectorStoreQueryResult,
+)
+
+_logger = logging.getLogger(__name__)
+
+DEFAULT_MMR_PREFETCH_FACTOR = 4.0
+
+
+class CassandraVectorStore(VectorStore):
+    """Cassandra Vector Store.
+
+    An abstraction of a Cassandra table with
+    vector-similarity-search. Documents, and their embeddings, are stored
+    in a Cassandra table and a vector-capable index is used for searches.
+    The table does not need to exist beforehand: if necessary it will
+    be created behind the scenes.
+
+    All Cassandra operations are done through the cassIO library.
+
+    Args:
+        session (cassandra.cluster.Session): the Cassandra session to use
+        keyspace (str): name of the Cassandra keyspace to work in
+        table (str): table name to use. If not existing, it will be created.
+        embedding_dimension (int): length of the embedding vectors in use.
+        ttl_seconds (Optional[int]): expiration time for inserted entries.
+            Default is no expiration.
+
+    """
+
+    stores_text: bool = True
+    flat_metadata: bool = True
```

Comment:
Just to confirm this is set properly, `flat_metadata` enforces the metadata to be a flat dict 
i.e.`key: str` -> `val: str | float | int`

And I'm guessing this is set because storing something like lists won't work in Cassandra?

Reply:
Well, at the moment the underlying metadata handling stores a blob, so deeper dicts would not be a problem. But we have an upcoming enhancement that handles indexes around these metadata fields, which makes it convenient to restrict to Dict[str, some-scalar-type] right away.
Thank you for asking!